### PR TITLE
SettingsProvider

### PR DESCRIPTION
**Pull request by EricJohnson327**
## Summary of the pull request
Implement SettingsProvider to display settings in Dev Home.

## PR checklist
- [ ] Closes #xxx
- [ ] Tests added/passed
- [ ] Documentation updated

### CODE REVIEW

File: src/GithubPlugin/Providers/SettingsUIController.cs [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+            Log.Logger()?.ReportDebug($"action: {action}");
+
+            switch (_settingsUI?.State)
+            {
+                case "SettingsPage":
+                    {
+                        Log.Logger()?.ReportDebug($"inputs: {inputs}");
+
+                        var currentNotificationsEnabled = LocalSettings.ReadSettingAsync<string>("NotificationsEnabled").Result ?? "true";
+                        await LocalSettings.SaveSettingAsync("NotificationsEnabled", currentNotificationsEnabled == "true" ? "false" : "true");
```
Comment (krschau) [inline src/GithubPlugin/Providers/SettingsUIController.cs:48]: Nit: make these strings variables that can be reused.

**Review by krschau — APPROVED** [review]

File: src/GithubPlugin/Providers/SettingsProvider.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft Corporation and Contributors
+// Licensed under the MIT license.
+
+using System.Xml.Linq;
+using GitHubPlugin.Client;
+using GitHubPlugin.DeveloperId;
+using GitHubPlugin.Helpers;
+using Microsoft.Windows.DevHome.SDK;
+using Octokit;
```
Comment (dhoehna) [inline src/GithubPlugin/Providers/SettingsProvider.cs:9]: Ocktokit?

Comment (EricJohnson327) [reply src/GithubPlugin/Providers/SettingsProvider.cs:9]: Fixed.

File: src/GithubPlugin/Providers/SettingsUIController.cs (hunk 1 above)
Comment (EricJohnson327) [reply src/GithubPlugin/Providers/SettingsUIController.cs]: Fixed.

File: src/GithubPlugin/Providers/SettingsProvider.cs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+// Licensed under the MIT license.
+
+using GitHubPlugin.Helpers;
+using Microsoft.Windows.DevHome.SDK;
+
+namespace GitHubPlugin.Providers;
+
+public class SettingsProvider : ISettingsProvider
+{
+    public SettingsProvider()
```
Comment (dhoehna) [inline src/GithubPlugin/Providers/SettingsProvider.cs:11]: NIT: Remove empty constructor.

File: src/GithubPlugin/Helpers/LocalSettings.cs [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+using Windows.Storage;
+
+namespace GitHubPlugin.Helpers;
+public static class LocalSettings
+{
+    private static readonly string _applicationDataFolder = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData), "DevHome/ApplicationData");
+    private static readonly string _localSettingsFile = "LocalSettings.json";
+
+#pragma warning disable IDE0044 // Add readonly modifier
+    private static IDictionary<string, object>? _settings;
```
Comment (dhoehna) [inline src/GithubPlugin/Helpers/LocalSettings.cs:18]: Why disable the warning?  I thought readonly can be applied to static.

File: src/GithubPlugin/Providers/SettingsUIController.cs [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+namespace GitHubPlugin.Providers;
+internal class SettingsUIController : IExtensionAdaptiveCardSession
+{
+    private static readonly string _notificationsEnabledString = "NotificationsEnabled";
+
+    private IExtensionAdaptiveCard? _settingsUI;
+    private static readonly SettingsUITemplate _settingsUITemplate = new ();
+
+    public SettingsUIController()
```
Comment (dhoehna) [inline src/GithubPlugin/Providers/SettingsUIController.cs:18]: NIT: Remove empty constructor.

File: src/GithubPlugin/Providers/SettingsProvider.cs (hunk 3 above)
Comment (EricJohnson327) [reply src/GithubPlugin/Providers/SettingsProvider.cs:11]: Fixed

File: src/GithubPlugin/Providers/SettingsUIController.cs (hunk 5 above)
Comment (EricJohnson327) [reply src/GithubPlugin/Providers/SettingsUIController.cs:18]: Fixed

File: src/GithubPlugin/Helpers/LocalSettings.cs (hunk 4 above)
Comment (EricJohnson327) [reply src/GithubPlugin/Helpers/LocalSettings.cs:18]: Removed.  This was leftover during development before I was actually changing _settings.

**Pull request merged**